src/App.js

```diff
@@ -0,0 +1,30 @@
+import { Route, Routes } from 'react-router-dom';
+import './App.css';
+import { useDispatch } from 'react-redux';
+
+import { fetchMissions } from './slices/MessionSlice';
+import MyProfile from './components/MyProfile';
+import Rockets from './components/Rockets';
+import NavBar from './components/NavBar';
+import MessionList from './components/MessionList';
+import { getRockets } from './slices/RocketSlice';
+
+function App() {
+  const dispatch = useDispatch();
+  dispatch(getRockets());
+  dispatch(fetchMissions());
```

Comment:
- [ ] According to the project requirements, `rockets` is the default view, so its data would be fetched when the application starts (fetching it in `App` is good 👏 ). However, the `missions` data should only be fetched (once) when a user navigates to the Missions section. Kindly move `dispatch(fetchMissions())` to the missions component (please make sure to put it in `useEffect()` with an empty dependency array to make sure it's fetched once 💯 )

Reply:
Thank you for your feedback, 
but about the mission's data: if I fetched it inside the missions component, then it will be fetched each time the component rerenders (the user navigates the component).
And for the `useEffect`: I don't think I need it because the `App` component will never rerender (it has no State).

I hope I hear from you soon @ha-manel 👍 
Cheers 😃

Reply:
Hi @Abdessittir👋 So since it's in the requirements to fetch the missions only when the user actually goes to the missions page (for optimization reasons) I think the only solution here is to fetch the data inside the missions component itself since fetching them in `App` means the data will be fetched before the user goes to the missions page and the user might never visit that page so it was fetched for no reason. 
To answer this (if I fetched it inside the missions component, then it will be fetched each time the component rerenders): Dispatching the action inside `useEffect` with an empty dependency array ensures that the fetching happens only once when the component is rendered the first time, please check [this](https://www.w3schools.com/react/react_useeffect.asp) for more information and let me know if you still have more concerns 😄

Reply:
Hi @ha-manel,
Unfortunately, I can't add the `useEffect` inside the missions component (even with an empty array), because React router will add the component from scratch (not just rerender it)
Thank 👍